x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/TrainedModelAssignment.java

```diff
@@ -533,6 +533,9 @@ public AssignmentState calculateAssignmentState() {
             if (assignmentState.equals(AssignmentState.STOPPING)) {
                 return assignmentState;
             }
+            if (taskParams.getNumberOfAllocations() == 0) {
+                return AssignmentState.STARTED;
```

Comment:
There are code paths where you can get into state of `numberOfAllocations=0` and `assignmentState=STARTING`. I think that can happen when the scheduler moves jobs around, which triggers the state to be recalculated here.

This causes issues, because assignments in a `STARTING` state cannot be updated, and an assignment with `numberOfAllocations=0` won't get out of that state.